wpilibc/src/main/native/cpp/DriverStation.cpp

```diff
@@ -509,11 +509,15 @@ bool DriverStation::IsFMSAttached() {
   return controlWord.fmsAttached;
 }
 
-std::string DriverStation::GetGameSpecificMessage() {
+std::optional<std::string> DriverStation::GetGameSpecificMessage() {
   HAL_MatchInfo info;
   HAL_GetMatchInfo(&info);
-  return std::string(reinterpret_cast<char*>(info.gameSpecificMessage),
-                     info.gameSpecificMessageSize);
+  if (info.gameSpecificMessage) {
```

Comment:
Is it actually null or is it zero-length?

Reply:
:thinking:

Reply:
My guess is 0 length based off the compile error

Reply:
No, this isn't the sort of thing you should be guessing at, it can be empirically (or otherwise) determined.

Especially When making a PR for a public project that is used by others, we probably shouldn't be guessing at things that are testable and/or knowable. There are certainly things that one has to guess at because it isn't possible to know it, or it involves a judgement call of some kind. This isn't one of those sorts of things (after all, how can you be sure that you fixed the bug that the PR is trying to fix without testing it?), and recognizing when that's the case and doing the extra work when needed will make you a better developer in the long run.

I don't typically make these sorts of critiques, but I do recognize that you've made a lot of high quality contributions to WPILib and seem to be very capable, and so I hope this thought will positively impact you and anyone else who reads this conversation.

Reply:
I do plan to test this and as such probably should've marked this as a draft. I more meant guess it's the correct approach to take before testing it (ie test this approach first) but that's a very valid criticism and I definitely should've made that more clear, my apologies!